dialects/linux/dproc.c

```diff
@@ -1395,6 +1397,50 @@ process_id(idp, idpl, cmd, uid, pid, ppid, pgid, tid, tcmd)
 	return(0);
 }
 
+/* compare namespace of this lsof proces and the target process */
+
+int
+compare_namespace(ns)
+        char *ns;                /* namespace */
+{
+        char nspath_self[NS_PATH_LENGTH], nspath_target[NS_PATH_LENGTH];
+        char ns_self[NS_LENGTH], ns_target[NS_LENGTH];
+        ssize_t n_self, n_target;
+
+        sprintf(nspath_self, "/proc/self/ns/%s", ns);
```

Comment:
May overflow the buffer if `ns` is sufficiently long. Prefer `snprintf` and check return value.

Reply:
Since '/proc/self/ns/[ns]' is unlikely to be longer than NS_PATH_LENGTH, which is defined as 100, I'm afraid there's no need to.

Reply:
Depending on the threat model you should …

There's nothing stopping an attacker from mounting a malicious procfs over normal proc and generating excessively large names inside that directory …

Also it's just bad style leaving your assumptions unchecked when doing so can result in bugs like buffer overflows (which in this case could potentially allow for arbitrary code execution).